src/sources/serum.ts

```diff
@@ -47,29 +41,21 @@ export class SerumMarketSource implements MarketSource {
   constructor(
     connection: Connection,
     tokenMap: TokenMap,
-    marketInfoMap: SerumMarketInfoMap
+    markets: ISerumMarketInfo[]
   ) {
     this.connection = connection;
     this.tokenMap = tokenMap;
-    this.marketInfoMap = marketInfoMap;
-
-    this.marketByMint = Object.values(this.tokenMap).reduce((map, token) => {
-      const marketInfo =
-        marketInfoMap[`${token.symbol}/USDC`] ??
-        marketInfoMap[`${token.symbol}/USDT`];
-      if (marketInfo && !marketInfo.deprecated) {
-        map.set(token.address, {
-          marketInfo,
-          tokenInfo: token,
-        });
-      }
-
-      return map;
-    }, new Map<string, ISerumMarketTokenInfo>());
-
-    this.marketKeys = [...this.marketByMint.values()].map((m) => {
-      return m.marketInfo.address;
-    });
+    this.markets = markets;
+
+    this.marketKeys = this.markets.reduce<string[]>(
+      (keys, { address, baseMintAddress, deprecated }) => {
+        if (!deprecated && this.tokenMap[baseMintAddress]) {
```

Comment:
I wonder if we should have a sanity check here that we lost from the previous logic which checks to ensure the quote mint is USDC or USDT.

Reply:
To do this, we would need to expose the `quoteMintAddress` in the `serum-markets` repo. Would have to pull all the Serum market info again to do this.

Reply:
Why would we need to do that? Couldn't it just be an inner check after we find the market using the `baseMintAddress` and check that the `name` contains USDC or USDT?

Reply:
I could expose it from the `getMarketPrices` method and do a check to see if it's either the `USDC` or `USDT` mint. Would that work?

Reply:
See second commit, added the requested functionality

Reply:
Don't you already have access to the `name` field right here since you're dealing with `ISerumMarketInfo` and are accessing `deprecated` etc already

Reply:
I want to avoid doing symbol comparison, that caused this issue in the first place. I added a comparison check against USDC and USDT mints.

Reply:
Fair enough. USDC and USDT bit different and less likely to be duped though :+1: